kuma/javascript/src/header/header.jsx

```diff
@@ -153,18 +92,112 @@ type Props = {
 export default function Header(props: Props): React.Node {
     const locale = getLocale();
 
-    function fixurl(url) {
-        // The "Report a content issue" menu item has a link that requires
-        // the document slug, so we work that in here. If there is no
-        // document data, then we're on the home page and just use '/locale'
-        let slug = props.document ? props.document.slug : `/${locale}`;
-
-        url = url.replace('{{SLUG}}', encodeURIComponent(slug));
-        if (!url.startsWith('https://')) {
-            url = `/${locale}/docs/${url}`;
-        }
-        return url;
-    }
+    // The menus array includes objects that define the set of
+    // menus displayed by this header component. The data structure
+    // is defined here during rendering so that the gettext() calls
+    // happen after the string catalog is available. And we're using
+    // useMemo() so that localization only happens when the locale
+    // changes.
+    const menus = useMemo(
```

Comment:
Perhaps this is nitpickery, if so I appologies in advance. 
The `<Header/>` component is used on every page mount, right. So the `props.searchQuery` is potentially different and internally (is that called a side-effect) the `locale` is different. And obviously `props.document` can change too. 

I noticed that the `<Header/>` component renders twice on every page. I put in some `console.log` of `locale` and `props.searchQuery` and it looks like this on every page click:
<img width="815" alt="Screen Shot 2019-06-27 at 3 41 01 PM" src="https://user-images.githubusercontent.com/26739/60295548-ff57df80-98f1-11e9-9332-91a48f7fe146.png">

(PS. I'm still grappling with why the prop specifies `searchQuery?: string` but it's allowed to be `undefined`)

Anyway, we could just memoize the whole component since the only inputs are easy to compare. E.g.

```
// Totally made up
export const Header = React.memo((props: Props): React.Node => {
  // will only render once per props passed
})
```
That way, the whole component would be memoized. And not just one smaller part of it inside. 
Immediate effect is that the whole component would only render *once* per page. 

I don't think we could get away from rendering the component for *every different* page because the URL to github issues depends on the document.

Reply:
That might be a good idea. I'm not sure why we're getting two renders. But it is unrelated to this localization issue, so I'd rather not tackle it as part of this PR.